src/Plugin/GraphQL/DataProducer/Field/EntityReferenceRevisions.php

```diff
@@ -0,0 +1,216 @@
+  ) {
+    parent::__construct($configuration, $pluginId, $pluginDefinition);
+    $this->entityTypeManager = $entityTypeManager;
+    $this->entityRevisionBuffer = $entityRevisionBuffer;
+  }
+
+  /**
+   * Resolves entity reference revisions for a given field of a given entity.
+   *
+   * May optionally respect the entity bundles and language.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity.
+   * @param string $field
+   *   The field of a given entity to get entity reference revisions for.
+   * @param string|null $language
+   *   Optional. Language to be respected for retrieved entities.
+   * @param array|null $bundles
+   *   Optional. List of bundles to be respected for retrieved entities.
+   * @param bool $access
+   *   Whether check for access or not. Default is true.
+   * @param \Drupal\Core\Session\AccountInterface|null $accessUser
+   *   User entity to check access for. Default is null.
+   * @param string $accessOperation
+   *   Operation to check access for. Default is view.
+   * @param \Drupal\graphql\GraphQL\Execution\FieldContext $context
+   *   The caching context related to the current field.
+   *
+   * @return \GraphQL\Deferred|null
+   *   A promise that will return entities or NULL if there aren't any.
+   */
+  public function resolve(EntityInterface $entity, string $field, ?string $language, ?array $bundles, ?bool $access, ?AccountInterface $accessUser, ?string $accessOperation, FieldContext $context): ?Deferred {
+    if (!$entity instanceof FieldableEntityInterface || !$entity->hasField($field)) {
+      return NULL;
+    }
+
+    $definition = $entity->getFieldDefinition($field);
+    if ($definition->getType() !== 'entity_reference_revisions') {
+      return NULL;
+    }
+
+    $definition = $entity->getFieldDefinition($field);
+    $type = $definition->getSetting('target_type');
+    if (($values = $entity->get($field)) && $values instanceof EntityReferenceFieldItemListInterface) {
+      $vids = array_map(function ($value) {
+        return $value['target_revision_id'];
+      }, $values->getValue());
+
+      $resolver = $this->entityRevisionBuffer->add($type, $vids);
+      return new Deferred(function () use ($type, $language, $bundles, $access, $accessUser, $accessOperation, $resolver, $context) {
+        $entities = $resolver() ?: [];
+
+        // Get the language version of entities.
+        if (isset($language)) {
+          $entities = array_map(function (EntityInterface $entity) use ($language) {
+            if ($language != $entity->language()->getId() && $entity instanceof TranslatableInterface) {
+              return $entity->getTranslation($language);
+            }
+
+            return $entity;
+          }, $entities);
+        }
+
+        $entities = array_filter($entities, function (EntityInterface $entity) use ($bundles, $access, $accessOperation, $accessUser, $context) {
+          if (isset($bundles) && !in_array($entity->bundle(), $bundles)) {
```

Comment:
This filtering should happen before the language loading. Otherwise you might end up loading translations for entities you are then filtering out in the next step.

Reply:
I actually noticed :D fixing it right now, but, damn, you were faster :D

Reply:
fixed now, please recheck

Reply:
Sweet.